pkg/textparse/lex.l

```diff
@@ -65,65 +54,31 @@ M     [a-zA-Z_:]
-                                      l.offsets = append(l.offsets, l.i)
-
-<lstateLName>{L}({L}|{D})*            l.state = lstateLEq
-                                      l.offsets = append(l.offsets, l.i)
-
-<lstateLEq>[ \t]*=                    l.state = lstateLValue
-
-<lstateLValue>[ \t]+
-<lstateLValue>\"                      l.state = lstateLValueIn
-                                      l.offsets = append(l.offsets, l.i)
-<lstateLValueIn>(\\.|[^\\"])*\"       l.state = lstateLabels
-                                      if !utf8.Valid(l.b[l.offsets[len(l.offsets)-1]:l.i-1]) {
-                                        l.err = fmt.Errorf("invalid UTF-8 label value")
-                                        return -1
-                                      }
-                                      l.offsets = append(l.offsets, l.i-1)
-
-<lstateValue>[ \t]+                   l.vstart = l.i
-<lstateValue>(NaN)                    l.val = math.Float64frombits(value.NormalNaN)
-                                      l.state = lstateTimestamp
-
-<lstateValue>[^\n \t\r]+              // We don't parse strictly correct floats as the conversion
-                                      // repeats the effort anyway.
-                                      l.val, l.err = strconv.ParseFloat(yoloString(l.b[l.vstart:l.i]), 64)
-                                      if l.err != nil {
-                                          return -1
-                                      }
-                                      l.state = lstateTimestamp
-
-<lstateTimestamp>[ \t]+               l.tstart = l.i
-<lstateTimestamp>{D}+                 ts, err := strconv.ParseInt(yoloString(l.b[l.tstart:l.i]), 10, 64)
-                                      if err != nil {
-                                          l.err = err
-                                          return -1
-                                      }
-                                      l.ts = &ts
-<lstateTimestamp>[\r\n]+              l.nextMstart = l.i
-                                      return 1
-<lstateTimestamp>\0                   return 1
+\0                                    return tEOF
+\r?\n                                 l.state = sInit; return tLinebreak
+<*>[ \t]+                             return tWhitespace
+
+#\ HELP\                              l.state = sMeta1; return tHelp
+#\ TYPE\                              l.state = sMeta1; return tType
+#.?.?.?.?.?.?                         // Hack to make comments work that start with a prefix
+                                      // of the above two rules. The generator does not
+                                      // seem to do the right thing here.
+                                      l.state = sComment
+<sComment>[^\r\n]*                    l.state = sInit; return tComment
+<sMeta1>{M}({M}|{D})*                 l.state = sMeta2; return tMName
+<sMeta2>[^\n\r]+                      l.state = sInit; return tText
```

Comment:
How are you handling escaping here?

Reply:
Escaping of what?

Reply:
The value of HELP can contain `\n` meaning a literal newline, and `\\` meaning a single backslash.